### PR TITLE
Add Go

### PR DESCRIPTION
**Pull request by kevlarr**
There is nothing really clever about any of the strategies used, and there is nothing noteworthy about the organization at all. Woe is you for even reading this.

Mostly just trying to learn the *basics* of Go - which, given Go's complexity, pretty much covers 75% of the language. *zing*

### CODE REVIEW

File: go_/fourscore.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+func createPlayers() []models.Player {
+    first  := models.NewPlayer(reader, "First", "x")
+    second := models.NewPlayer(reader, "Second", "o")
+
+    return []models.Player{first, second}
+}
+
+func createBoard() models.Board {
+    return models.NewBoard()
```
Comment (kevlarr) [inline go_/fourscore.go:50]: Requirement was *supposed* to include custom board size, but I'm so... bored.. (eh?)... of spending time on that

Comment (kevlarr) [reply go_/fourscore.go:50]: I'm writing comments to myself on my own PR, how boring is that?

File: go_/models/board.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            return row, nil
+        }
+    }
+    return -1, errors.New("Column is full")
+}
+
+func (b *Board) Inspect(row, col int) (win bool, draw bool) {
+    win =
+        b.wonVertical(row, col) || b.wonHorizontal(row, col) ||
+        b.wonDiagonal(row, col) || b.wonAntidiagonal(row, col)
```
Comment (kevlarr) [inline go_/models/board.go:47]: I bet there's some crazy way to bit-shift the checking for wins, but I'm not smart. Plus, I code in bed after bedtime.

Comment (justonia) [reply go_/models/board.go:47]: Computers are fast so we don't have to be

File: go_/models/player.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        if err != nil { continue }
+
+        col--
+        row, err = board.Receive(col, p.token)
+
+        if err == nil { break }
+    }
+
+    return
+}
```
Comment (kevlarr) [inline go_/models/player.go:40]: These things probably shouldn't be implemented on `*Player`, it was just fun to play around with the module/subpackage system

**Comment by justonia** [comment]
It's actually funny to me how foreign Go feels after being in c# land for soooooo long now. I actually don't mind the rust syntax from the brief looks I took at that one.

**Comment by kevlarr** [comment]
@justonia Ignoring the addition of manual management and the resulting "single owner" struggle and reference lifetime hell, Rust feels a lot like Go to me (syntax and otherwise)

**Pull request merged**